File: 0946-validate-stack-sequences/0946-validate-stack-sequences.py
class Solution:
    def validateStackSequences(self, pushed: List[int], popped: List[int]) -> bool:
        q = deque()
        
        push_size = len(pushed)
        pop_size = len(popped)
        popCounter = 0
        
        for i in range(push_size):
            q.append(pushed[i])
            while q and popped[popCounter] == q[-1] and popCounter < pop_size:
                q.pop()
                popCounter += 1
        
        if q:
            return False
        else:
            return True

File: 0946-validate-stack-sequences/0946-validate-stack-sequences.cpp
class Solution {
public:
    bool validateStackSequences(vector<int>& pushed, vector<int>& popped) 
    {
        stack < int > st;
        
        int popCounter = 0;
        
        int push_size = pushed.size();
        int pop_size = popped.size();
        
        for (int i = 0 ; i < push_size ; ++i)
        {
            st.push(pushed[i]);
            while(!st.empty() && popped[popCounter] == st.top() && popCounter < pop_size)
            {
                st.pop();
                popCounter++; 
            }
        }
        if (st.empty())
        {
            return true;
        }
        else
        {
            return false;
        }
        
    }
};